StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -315,7 +315,9 @@ extension PaymentSheet {
                         with: authenticationContext,
                         completion: { actionStatus, _, error in
                             paymentHandlerCompletion(actionStatus, error)
-                            linkAccount?.logout()
+                            if actionStatus == .succeeded {
+                                linkAccount?.logout()
```

Comment:
Yes, otherwise if a payment is cancelled (e.g. during a 3DS2 challenge) or fails (for insufficient funds), the user would need to log in again before trying with another card.

Reply:
Makes sense!